src/module/krf.c

```diff
@@ -330,3 +340,63 @@ static ssize_t log_faults_file_write(struct file *f, const char __user *ubuf, si
   *off = buflen;
   return buflen;
 }
+
+static ssize_t targeting_file_read(struct file *f, char __user *ubuf, size_t size, loff_t *off) {
+  char buf[KRF_PROCFS_MAX_SIZE + 1] = {0};
+  size_t buflen = 0;
+  size_t offset = 0;
+  size_t current_mode;
+  for (current_mode = 0; current_mode < KRF_T_NUM_MODES; current_mode++) {
+    if ((krf_target_options.mode_mask & (1 << current_mode)) && (offset < KRF_PROCFS_MAX_SIZE)) {
+      offset += sprintf(buf + offset, "%lu %u\n", current_mode,
+                        krf_target_options.target_data[current_mode]);
+    }
+  }
+  buflen = strnlen(buf, KRF_PROCFS_MAX_SIZE);
+
+  if (*off > 0 || size < buflen) {
+    return 0;
+  }
+
+  if (copy_to_user(ubuf, buf, buflen)) {
+    return -EFAULT;
+  }
+
+  *off = buflen;
+  return buflen;
+}
+
+static ssize_t targeting_file_write(struct file *f, const char __user *ubuf, size_t size,
+                                    loff_t *off) {
+  char buf[KRF_PROCFS_MAX_SIZE + 1] = {0};
+  size_t buflen = 0;
+  krf_target_mode_t mode;
+  unsigned int data;
+
+  if (size > KRF_PROCFS_MAX_SIZE) {
+    size = KRF_PROCFS_MAX_SIZE;
+  }
+
+  if (copy_from_user(buf, ubuf, size)) {
+    return -EFAULT;
+  }
+
+  if (sscanf(buf, "%u %u", &mode, &data) != 2) {
+    return -EINVAL;
+  }
+
+  if ((mode == 0) && (data == 0)) { // If both arguments are zero, remove all targeting
```

Comment:
Nit: Let's add a `printk` in this case as well, like the syscall table clearing case.